pages/api/autocomplete.ts

```diff
@@ -65,24 +121,26 @@ function bfsRecursion(queue: QueueItem[]) {
   let returnData = false;
 
   //# of characters matched
-  const nodeCharacters = graph.getNodeAttribute(queueItem?.node, 'c');
+  const nodeCharacters = graph.getNodeAttribute(queueItem?.data?.node, 'c');
   let matches = 0;
   while (
     matches < nodeCharacters.length &&
-    queueItem?.characters?.[0] === nodeCharacters[0]
+    matches < queueItem?.data?.characters?.length
```

Comment:
Does this work where `characters` is undefined/nil?

Reply:
I don't think `characters` will ever be undefined. I can add a check (`typeof queueItem?.data?.characters?.length !== 'undefined'`) that will discard that search result if it ever is. Do you think that's a good idea?

Reply:
Yeah, it makes sense that it shouldn't be undefined, though just wanted to be sure that couldn't happen. If we know that `data` can't be undefined then this is probably fine as written.